### PR TITLE
Doc: Switch to Dokkatoo

### PR DESCRIPTION
**Pull request by martinbonnin**
Use https://github.com/adamko-dev/dokkatoo that is configuration cache compatible and cache relocatable

The workers use *a lot* of memory so depending the machine, we can't run too many in parallel. I set the number of parallel workers to 4 for now. 🤞 that it works on GA runners...

### CODE REVIEW

**Review by BoD — APPROVED** [review]
👍

File: build-logic/src/main/kotlin/Publishing.kt [hunk 1]
```diff
@@ -49,26 +50,52 @@ fun Project.configurePublishing() {
...
+      suppressedFiles.from(file("src/main/kotlin/it/suppressedByPath"))
+      perPackageOption {
+        matchingRegex.set(".*internal")
+        suppress.set(true)
+      }
+    }
+
+    dokkatooPublications.configureEach {
+      suppressObviousFunctions.set(true)
+      suppressObviousFunctions.set(false)
```
[the remark this answers is not in the document]
Comment (martinbonnin) [reply build-logic/src/main/kotlin/Publishing.kt:90]: Nope, bad copy paste from the documentation, thanks 🙏

**Comment by martinbonnin** [comment]
On hold for https://github.com/adamko-dev/dokkatoo/pull/64

**Comment by martinbonnin** [comment]
Closing this one for now, will revist later

**Pull request closed without merging**